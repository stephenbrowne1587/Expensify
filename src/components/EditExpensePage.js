import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import {startEditExpense,
        startRemoveExpense
        } from "../actions/expenses";
import Header from "./Header";

export class EditExpensePage extends React.Component {
  onRemove = () => {
    // this.props.dispatch(removeExpense({id: this.props.expense.id}))
    this.props.startRemoveExpense({id: this.props.expense.id});
    this.props.history.push("/dashboard");
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/dashboard");
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button 
            className="button button--secondary" 
            onClick={this.onRemove}>
            Remove Expense
          </button>  
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => 
  expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
  startEditExpense: (id, expense) => dispatch(startEditExpense( id,expense))
});

export default connect(
  mapStateToProps, mapDispatchToProps)( EditExpensePage);