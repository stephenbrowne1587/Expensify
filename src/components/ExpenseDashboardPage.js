import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpenseSummry from "./ExpensesSummary";


const ExpenseDashboardPage = () => (
  <div>
    <ExpenseSummry/>
    <ExpenseListFilters/>
    <ExpenseList/>
  </div>
);
export default ExpenseDashboardPage;