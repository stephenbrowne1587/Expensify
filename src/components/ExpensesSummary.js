import React from "react";
import {connect} from "react-redux";
import selectExpenses from "../selectors/expenses";
import getExpenseTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpenseSummary = (props) => (
  
  <div>
  {
    props.expenses.length > 0 &&
      <p>
        Viewing {props.expenses.length
        } {props.expenses.length === 1 ? "expense" : "expenses"} totalling {
          numeral(getExpenseTotal(props.expenses)/100).format("$0,0.00")}
      </p>
  }

  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);


