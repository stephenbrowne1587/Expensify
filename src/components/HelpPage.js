import React from "react";

const HelpPage = () => (
  <div>
      This is the HelpPage
  </div>
);

export default HelpPage;