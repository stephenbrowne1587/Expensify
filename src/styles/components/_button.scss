.button {
  background-color: $blue;
  color: white;
  display: inline-block;
  padding: $s-size;
  font-family: Arial, Helvetica, sans-serif;
  border: none;
  font-size: $font-size-large;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888
}