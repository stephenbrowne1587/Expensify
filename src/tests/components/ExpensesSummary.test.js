import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should render expense summary with multiple expenses", () => {
  const wrapper = shallow(<ExpenseSummary expenses={expenses}/>)
  expect(wrapper).toMatchSnapshot();
});

test("Should render expenses summary with one expense", () => {
  const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>)
  expect(wrapper).toMatchSnapshot();
});

