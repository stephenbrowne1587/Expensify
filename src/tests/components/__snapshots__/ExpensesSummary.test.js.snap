// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenses summary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </p>
</div>
`;

exports[`should render expense summary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $1,141.95
  </p>
</div>
`;
